src/WebJobs.Script.WebHost/Controllers/KeysController.cs

```diff
@@ -87,17 +88,17 @@ private IHttpActionResult GetKeysResult(IDictionary<string, string> keys)
             return Ok(keyResponse);
         }
 
-        private IHttpActionResult PutKey(string keyName, Key key, string functionName = null)
+        private async Task<IHttpActionResult> PutKey(string keyName, Key key, string functionName = null)
         {
             if (key?.Value == null)
             {
                 return BadRequest("Invalid key value");
             }
 
-            return AddOrUpdateFunctionSecret(keyName, key.Value, functionName);
+            return await AddOrUpdateFunctionSecret(keyName, key.Value, functionName);
```

Comment:
Yeah, this was intentionally left this way (actually went back on some that I had changed), will add it for consistency, but tired of that :)